cache.go

```diff
@@ -23,6 +23,8 @@ type Interface[K comparable, V any] interface {
 	Keys() []K
 	// Delete deletes the item with provided key from the cache.
 	Delete(key K)
+	// Len returns the number of items in the cache.
+	Len() int
```

Comment:
Should add method of the Cache struct because this is breaking change.

I think good like below:

Len() int { return len(c.Keys()) }

Reply:
Do you mean that this my change below:
```
func (c *Cache[K, V]) Len() int {
	c.mu.Lock()
	defer c.mu.Unlock()
	return c.cache.Len()
}
```
should be changed to:
```
func (c *Cache[K, V]) Len() int {
	return len(c.Keys())
}
```

Reply:
@norri Yes!

Reply:
Could you measure the performance?
And also please implement all cache policies not just simple 🙏

Reply:
I can try to test the performance. All other policies implement already `Len()` so no changes needed for those.

Reply:
I added benchmark test for both implementations and using just Len() is quite match faster:
```
$ go test -benchmem -bench BenchmarkLenWithKeys ./policy/simple
goos: darwin
goarch: amd64
pkg: github.com/Code-Hex/go-generics-cache/policy/simple
cpu: Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
BenchmarkLenWithKeys-8           3777514               310.7 ns/op           104 B/op          3 allocs/op
PASS
ok      github.com/Code-Hex/go-generics-cache/policy/simple     1.640s
```

```
$ go test -benchmem -bench BenchmarkJustLen ./policy/simple
goos: darwin
goarch: amd64
pkg: github.com/Code-Hex/go-generics-cache/policy/simple
cpu: Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
BenchmarkJustLen-8      1000000000               0.3075 ns/op          0 B/op          0 allocs/op
PASS
```